deliver/lib/deliver/upload_price_tier.rb

```diff
@@ -21,7 +21,7 @@ def upload(options)
       # Need to get prices from the app's relationships
       # Prices from app's relationship doess not have price tier so need to fetch app price with price tier relationship
       app_prices = app.prices
-      if app_prices.first
+      if app_prices && app_prices.first
```

Comment:
This can be shortened to the following:
```suggestion
      if app_prices&.first
```
However, I don't think it is the correct fix. `app.prices` being `nil` seems to be due to the change in #21187, but if that's indeed the case, `app_prices` will always be `nil`, so you end up always going in the `else` branch of this `if`, even if App Store Connect does have prices.